azure-client-runtime/src/main/java/com/microsoft/azure/AzureAsyncOperationPollStrategy.java

```diff
@@ -100,6 +103,8 @@ public void updateFrom(HttpResponse httpPollResponse) throws IOException {
                                         pollingSucceeded = SUCCEEDED.equalsIgnoreCase(provisioningState);
                                         clearDelayInMilliseconds();
                                     }
+                                } else if (httpPollResponse.statusCode() == 200) {
+                                    throw new CloudException("Response does not contain a valid body", httpPollResponse, null);
```

Comment:
Should this be a Single.error() instead of a throw?

Reply:
I think it has the same effect but would be cleaner to Single.error here.